tmux/paths.go

```diff
@@ -0,0 +1,35 @@
+package tmux
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// alternatePath returns an altnerate string that should be check when doing
+// path based comparisons.
+func alternatePath(s string) (altPath string) {
+	// If the path is absolute, there is no alternate path.
+	if filepath.IsAbs(s) {
+		return ""
+	}
+
+	// If the path starts with a ~ it's likely relative to the home directory.
+	if strings.HasPrefix(s, "~/") {
+		homeDir, err := os.UserHomeDir()
+		if err == nil {
+			altPath = filepath.Join(homeDir, strings.TrimPrefix(s, "~/"))
+		}
+	}
+
+	// If the path starts with a . it's likely relative to the current
+	// directory.
+	if strings.HasPrefix(s, ".") {
+		if a, err := filepath.Abs(s); err == nil {
+			altPath = a
+		}
+	}
+
+	// If we get to this point the path is likely a subdirectory.
+	return altPath
```

Comment:
Can we validate that this is a sub directory rather than just return it as a string?

Furthermore, if we get through all these checks should we return an empty string as no alternate path could be found?

Part of me thinks this could be added to the determine path logic instead of being it's own extra step.

Reply:
`AlternatePath` is used by the `GetSession` function in an attempt to account for situations where the user specifies a relative path while the tmux session has a fully qualified path. For those purposes it's not relevant if the path is valid, just if it can be used to locate the intended session.

The zero value of a string type in Go is an empty string, if this code can't determine an alternative path to update `altPath` that is what it will return. If you prefer I can add an extra check with an explicit return of an emptry string, though it would be functionally the same code.

This is an extra step to determine a potential alternative because there is no guarantee that it is the correct action. e.g. if the user is trying to connect to a session named `~/mysession` that is rooted in `/tmp` re-writing that name. explicity could cause unexpected / unwanted behavior.